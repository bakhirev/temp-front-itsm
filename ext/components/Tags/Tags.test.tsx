import React from 'react';
import { shallow } from 'enzyme';

import { Tags, IItemTag } from './';

const listItems: IItemTag[] = [
  { id: '1', label: 'Культура' },
  { id: '2', label: 'Кино' },
  { id: '3', label: 'Новости' },
  { id: '4', label: 'Музыка' },
];

const tagsRequiredProps = {
  onFilter: jest.fn(),
  items: listItems,
};

jest.mock('../common', () => ({
  DEFAULT_THEME: {},
  Z_INDEXES: {
    TOOLTIP: 777,
  },
}));

describe('Tags', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render component with items', () => {
    const wrapper = shallow(<Tags kind="red" {...tagsRequiredProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onClick when use click on component', () => {
    const wrapper = shallow(<Tags kind="blue" {...tagsRequiredProps} />);
    wrapper.childAt(0).simulate('click');
    const { onFilter } = tagsRequiredProps;
    expect(onFilter).toBeCalledTimes(1);
  });
});
