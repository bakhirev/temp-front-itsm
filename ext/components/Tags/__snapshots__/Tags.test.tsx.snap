// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags should render component with items 1`] = `
<styled.div>
  <styled.button
    key="1"
    kind="red"
    onClick={[Function]}
    tabIndex={1}
    theme={Object {}}
    tooltip="Культура"
  >
    <styled.div>
      Культура
    </styled.div>
  </styled.button>
  <styled.div />
  <styled.button
    key="2"
    kind="red"
    onClick={[Function]}
    tabIndex={1}
    theme={Object {}}
    tooltip="Кино"
  >
    <styled.div>
      Кино
    </styled.div>
  </styled.button>
  <styled.div />
  <styled.button
    key="3"
    kind="red"
    onClick={[Function]}
    tabIndex={1}
    theme={Object {}}
    tooltip="Новости"
  >
    <styled.div>
      Новости
    </styled.div>
  </styled.button>
  <styled.div />
  <styled.button
    key="4"
    kind="red"
    onClick={[Function]}
    tabIndex={1}
    theme={Object {}}
    tooltip="Музыка"
  >
    <styled.div>
      Музыка
    </styled.div>
  </styled.button>
</styled.div>
`;
